src/network/postgres/statement.cpp

```diff
@@ -0,0 +1,70 @@
+#include "network/postgres/statement.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "common/managed_pointer.h"
+#include "execution/executable_query.h"
+#include "parser/postgresparser.h"
+#include "traffic_cop/traffic_cop_util.h"
+#include "type/type_id.h"
+
+namespace terrier::network {
+
+Statement::Statement(std::string &&query_text, std::unique_ptr<parser::ParseResult> &&parse_result)
+    : Statement(std::move(query_text), std::move(parse_result), {}) {}
+
+Statement::Statement(std::string &&query_text, std::unique_ptr<parser::ParseResult> &&parse_result,
+                     std::vector<type::TypeId> &&param_types)
+    : parse_result_(std::move(parse_result)), param_types_(std::move(param_types)) {
```

Comment:
Seems that you need to move `query_text` into `query_text_` here?

Reply:
Whoops. Good catch.